reader.go

```diff
@@ -30,48 +30,58 @@ func (o *ReaderOptions) norm(name string) {
 
 // Reader reads data from a remote feed.
 type Reader struct {
-	remote *bfs.Object
-	opt    ReaderOptions
-	ctx    context.Context
-	num    int
+	io.Reader
 
-	br io.ReadCloser // bfs reader
-	cr io.ReadCloser // compression reader
-	fd FormatDecoder
+	remotes []*bfs.Object
+	closers []io.Closer
+	fd      FormatDecoder
+	opt     ReaderOptions
+
+	ctx context.Context
+	num int
 }
 
 // NewReader inits a new reader.
 func NewReader(ctx context.Context, remote *bfs.Object, opt *ReaderOptions) (*Reader, error) {
+	return MultiReader(ctx, []*bfs.Object{remote}, opt), nil
+}
+
+// MultiReader inits a new reader for multiple remotes.  Remotes are read sequentially as if concatenated.
+// Once all remotes are fully read, Read will return EOF.
+// MultiReader utilises a single format converter therefore all remotes must be same format.
+func MultiReader(ctx context.Context, remotes []*bfs.Object, opt *ReaderOptions) *Reader {
 	var o ReaderOptions
```

Comment:
I changed my mind, as we have multiple remotes, they may have different detected options (one compressed, another isn't or different formats). Please move opt normalisation and merging back into `streamReader`.